app/livechat/server/startup.js

```diff
@@ -24,7 +35,7 @@ Meteor.startup(() => {
 		}
 		const { _id: userId } = user;
 		const { servedBy: { _id: agentId } = {} } = room;
-		return userId === agentId;
+		return userId === agentId || (!room.open && hasPermission(user._id, 'view-livechat-closed-room-by-another-agent'));
```

Comment:
I think this permission name has been renamed [here](https://github.com/RocketChat/Rocket.Chat/pull/17580/files#diff-101fef96a9b5f18b5c1520e19e471d9eR115).
```suggestion
		return userId === agentId || (!room.open && hasPermission(user._id, 'view-livechat-room-closed-by-another-agent'));
```